src/machines/java/com/enderio/machines/common/blockentity/CrafterBlockEntity.java

```diff
@@ -0,0 +1,170 @@
+        return new CrafterMenu(this, inventory, containerId);
+    }
+
+    @Override
+    public MachineInventoryLayout getInventoryLayout() {
+        return MachineInventoryLayout
+            .builder()
+            .capacitor()
+            .setStackLimit(1)
+            .inputSlot(9, this::acceptSlotInput)
+            .setStackLimit(64)
+            .outputSlot(1)
+            .ghostSlot(9)
+            .previewSlot()
+            .build();
+    }
+
+    private boolean acceptSlotInput(int slot, ItemStack stack) {
+        return this.getInventory().getStackInSlot(slot + 10).sameItem(stack);
+    }
+
+    @Override
+    public void serverTick() {
+        processOutputBuffer();
+
+        Optional<ItemStack> opt = getRecipeResult();
+        if (opt.isPresent()) {
+            ItemStack result = opt.get();
+            this.getInventory().setStackInSlot(20, result);
+            if (shouldActTick() && hasPowerToCraft() && canMergeOutput(result) && outputBuffer.isEmpty()) {
+                craftItem();
+            }
+        } else {
+            this.getInventory().setStackInSlot(20, ItemStack.EMPTY);
+        }
+        super.serverTick();
+    }
+
+    private boolean shouldActTick() {
+        return canAct() && level.getGameTime() % ticksForAction() == 0;
+    }
+
+    private int ticksForAction() {
+        return 20;
+    }
+
+    private boolean hasPowerToCraft() {
+        return this.energyStorage.consumeEnergy(ENERGY_USAGE_PER_ITEM, true) > 0;
+    }
+
+    private void processOutputBuffer() {
+        if (outputBuffer.isEmpty()) {
+            return;
+        }
+        // clean buffer
+        outputBuffer.removeIf(ItemStack::isEmpty);
+
+        // output
+        if (canMergeOutput(outputBuffer.get(0))) {
+            var stack = getInventory().getStackInSlot(10);
+            if (stack.isEmpty()) {
+                getInventory().setStackInSlot(10, outputBuffer.get(0).copy());
+            } else if (stack.sameItem(outputBuffer.get(0))) {
+                stack.grow(1);
+            }
+            outputBuffer.remove(0);
+        }
+    }
+
+    private Optional<ItemStack> getRecipeResult() {
+        MachineInventory inv = this.getInventory();
+        int start = 11;
+        for (int i = 0; i < 9; i++) {
+            dummyCContainer.setItem(i, inv.getStackInSlot(start + i).copy());
+        }
+        Optional<CraftingRecipe> opt = getLevel().getRecipeManager().getRecipeFor(RecipeType.CRAFTING, dummyCContainer, getLevel());
+        if (opt.isPresent()) {
+            recipe = opt.get();
+            return Optional.of(recipe.assemble(dummyCContainer));
+        }
+        return Optional.empty();
+    }
+
+    private boolean canMergeOutput(ItemStack item) {
+        ItemStack output = this.getInventory().getStackInSlot(10);
+        return output.isEmpty() || (output.sameItem(item) && (output.getCount() + item.getCount() <= 64));
```

Comment:
do this check by calling ItemStack#isSameItemSameTags

Reply:
Just change `sameItem()` to `isSameItemSameTags()` ?

Reply:
Yep